PyDSS/pyPostprocessor/PostprocessScripts/AutomatedThermalUpgrade.py

```diff
@@ -692,143 +939,190 @@ def plot_feeder(self):
         self.plot_violations_counter+=1
 
     def export_line_DT_parameters(self):
-        self.originial_line_parameters = {}
+        self.original_line_parameters = {}
+        self.Linecodes_params = {}
         self.originial_xfmr_parameters = {}
-        dss.Lines.First()
-        while True:
-            ln_name = dss.Lines.Name()
-            ln_lc = dss.Lines.LineCode()
-            if ln_lc == '':
+
+        # creating dictionary with line parameters
+        if dss.Lines.First() == 1:  # first check if lines exist, to avoid segmentation fault
+            while True:
+                ln_name = dss.Lines.Name()
+                ln_lc = dss.Lines.LineCode()
                 ln_geo = dss.Lines.Geometry()
-                ln_lc = ln_geo
-            ln_len = dss.Lines.Length()
-            len_units = self.config["units key"][dss.Lines.Units()-1]
-            #len_units_alt = dss.Properties.Value("Units")
-            ln_phases = dss.Lines.Phases()
-            ln_b1 = dss.Lines.Bus1()
-            ln_b2 = dss.Lines.Bus2()
-            dss.Circuit.SetActiveBus(ln_b1)
-            kv_b1 = dss.Bus.kVBase()
-            dss.Circuit.SetActiveBus(ln_b2)
-            kv_b2 = dss.Bus.kVBase()
-            dss.Circuit.SetActiveElement("Line.{}".format(ln_name))
-            if kv_b1!=kv_b2:
-                raise InvalidParameter("To and from bus voltages ({} {}) do not match for line {}".format(kv_b2, kv_b1, ln_name))
-            self.originial_line_parameters[ln_name] = {"linecode":ln_lc,"length": ln_len,
-                                                       "num_phases": ln_phases,"line_kV": kv_b1,
-                                                       "length_unit":len_units}
-            if not dss.Lines.Next()>0:
-                break
-        self.write_to_json(self.originial_line_parameters, "Original_line_parameters")
+                if (ln_lc == '') & (ln_geo != ''):
+                    ln_lc = ln_geo
+
+                # if there are no line codes and geometries defined, create line code with line name
+                elif (ln_lc == '') & (ln_geo == ''):

```

Comment:
```suggestion
                elif ln_lc == '' and ln_geo == '':
```

Reply:
Okay, made suggested change in all places of occurrence